HomeWork_One/Quiz_Template/src/components/hooks/useTimer.tsx

```diff
@@ -0,0 +1,23 @@
+import {useEffect, useRef} from "react";
+
+export function useTimer(callback,enabled) {
+    const savedCallback = useRef<FC | null>(null)
+
+    useEffect(()=>{
+        savedCallback.current = callback
+    })
```

Comment:
this effect will be triggered every time
how often are you gonna override it? 🤔

Reply:
But I need it, because I should use this call back every time when i need to change state. Or maybe you can give me advice how can I do it more effective?

Reply:
to update state you have sibling useEffect
in this effect you just assigning a passed callback
I think here should be an empty array as a 2nd param